test/extended/builds/run_policy.go

```diff
@@ -310,6 +310,49 @@ var _ = g.Describe("[Feature:Builds][Slow] using build configuration runPolicy",
 			})
 		})
 
+		g.Describe("build configuration with Serial build run policy handling deletion", func() {
+			g.It("starts the next build immediately after running one is deleted", func() {
+				g.By("starting multiple builds")
+
+				bcName := "sample-serial-build"
+				for i := 0; i < 3; i++ {
+					_, _, err := exutil.StartBuild(oc, bcName)
+					o.Expect(err).NotTo(o.HaveOccurred())
+				}
+
+				buildWatch, err := oc.BuildClient().Build().Builds(oc.Namespace()).Watch(metav1.ListOptions{
+					LabelSelector: buildutil.BuildConfigSelector(bcName).String(),
+				})
+				defer buildWatch.Stop()
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				var deleteTime, deleteTime2 time.Time
+				for {
+					event := <-buildWatch.ResultChan()
```

Comment:
wrap this in a select timeout like the other TCs do, so we don't want forever if we don't get the next event.